src/SystemLink.cmake

```diff
@@ -10,15 +10,23 @@ function(target_include_system_directories target)
 
   foreach(scope IN ITEMS INTERFACE PUBLIC PRIVATE)
     foreach(lib_include_dirs IN LISTS ARG_${scope})
-      if(MSVC)
+      if(NOT MSVC)
         # system includes do not work in MSVC
         # awaiting https://gitlab.kitware.com/cmake/cmake/-/issues/18272#
         # awaiting https://gitlab.kitware.com/cmake/cmake/-/issues/17904
-        target_include_directories(${target} ${scope} ${lib_include_dirs})
+        set(_SYSTEM SYSTEM)
+      endif()
+      if(scope STREQUAL INTERFACE)
+        target_include_directories(
+          ${target}
+          ${_SYSTEM}
+          ${scope}
+          "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/${lib_include_dirs}>"
+          "$<INSTALL_INTERFACE:${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR}>")
```

Comment:
This seems to be needed to allow installing interface libraries. @lefticus Do you have a suggestion regarding this?